modules/compute-mig/variables.tf

```diff
@@ -66,11 +66,29 @@ variable "location" {
   type        = string
 }
 
+variable "minimal_action" {
+  description = "Minimal action to perform on instance during update. Can be 'NONE', 'REPLACE', 'RESTART' and 'REFRESH'."
+  type        = string # NONE | REPLACE | RESTART | REFRESH
+  default     = "NONE"
+}
+
+variable "most_disruptive_allowed_action" {
+  description = "Most disruptive action to perform on instance during update. Can be 'REPLACE, 'RESTART', 'REFRESH' or 'NONE'."
+  type        = string # REPLACE | RESTART | REFRESH | NONE
+  default     = "REPLACE"
+}
+
 variable "name" {
   description = "Managed group name."
   type        = string
 }
 
+variable "name_instance_config" {
```

Comment:
I would bundle all four stateful-related variables in a single `stateful_config` one, with four attributes: `name`, `destroy_state`, `group_policy`, `instance_policy`. I'm not entirely clear how do you select the instance to attach an instance policy too, by positional argument or instance name?